OneCommunityGlobal/HighestGoodNetworkApp
# Kaikanes lessonform update

# Description
Integrated backend post route to the front end page. Now users are able to post new lessons on the front end using the lessonForm. 

## Related PRS (if any):
This frontend PR is related to the  https://github.com/OneCommunityGlobal/HGNRest/pull/716 backend PR.

## Main changes explained:
- Create lessonActions
- Updated schema to hold the files and allowedRoles feilds
- Update lessonForm.jsx to dispatch action
- Imported toast to display success/error/info


## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to dashboard -> bmdashboard -> log in again -> bmdashboard/lessonform 
6. verify function submitting lessonform creates it in the backend
7. also check project specific form
8. go to bmdashboard -> select a project from the dropdown -> click 'Go to Project Dashboard' -> click 'Lessons' button
9. verify this functions the same too

## Screenshots or videos of changes:
bmdashboard/lessonform/

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114305309/6a0d58d7-4967-4f71-90fe-57643f24f2a1

bmdashboard/lessonform/:lessonId


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114305309/8e3f3837-0b82-430b-8e2f-d9e5a6d3322e


## Note:
The bmdashboard styles seem to be messed up. (unrelated to this PR)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114305309/8f6cce24-3a73-4c02-8430-46357cab916a)

Need auth token and set header to see backend in insomnia/postman

GRAB TOKEN IN APP FRONT END LOCAL STORAGE
<img width="270" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/dda410e3-9308-442c-898d-810e40dc5e92">
 HEADER (I use insomnia)
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/0562e588-3448-4e6a-8cf2-9ceb6e372d04)
